src/media/local-video-track.ts

```diff
@@ -0,0 +1,130 @@
+import { logger } from '../util/logger';
+import { LocalTrack } from './local-track';
+
+export enum FacingMode {
+  user = 'user',
+  environment = 'environment',
+}
+
+export enum OptimizationMode {
+  motion = 'motion',
+  detail = 'detail',
+}
+
+export type ConstrainLong = {
+  min?: number;
+  max?: number;
+  ideal?: number;
+  exact?: number;
+};
+
+export type VideoEncoderConfig = {
+  bitRateMin?: number;
+  bitrateMax?: number;
+  frameRate?: number | ConstrainLong;
+  height?: number | ConstrainLong;
+  width?: number | ConstrainLong;
+  aspectRatio?: number;
+};
+
+/**
+ * A class to map resolution with video constraints.
+ */
+export const staticVideoEncoderConfig = {
```

Comment:
type name should start with upper case letter and it would be better to specify the type here:
```
export type StaticVideoEncoderConfig: {[key: string]: VideoEncoderConfig}
```